doc/source/whatsnew/v0.25.0.rst

```diff
@@ -38,6 +38,7 @@ Other Enhancements
 - :class:`RangeIndex` has gained :attr:`~RangeIndex.start`, :attr:`~RangeIndex.stop`, and :attr:`~RangeIndex.step` attributes (:issue:`25710`)
 - :class:`datetime.timezone` objects are now supported as arguments to timezone methods and constructors (:issue:`25065`)
 - :meth:`DataFrame.query` and :meth:`DataFrame.eval` now supports quoting column names with backticks to refer to names with spaces (:issue:`6508`)
+- :meth:`DataFrame.to_csv` and :meth:`Series.to_csv` now support dicts as ``compression`` argument with key ``'method'`` being the compression method and others as additional compression options when the compression method is ``'zip'``. (:issue:`26023`)
```

Comment:
It's hinted at but not explicitly stated that the `dict` only works when the `method` key is `zip` right? If so can we clarify that via the docstrings and make sure we raise if another method is supplied? I think current implementation just swallows other kwargs for other compression methods